### DISCUSSION TITLE
Merging props with individual passthroughs

### DISCUSSION THREAD

**Discussion by willbdv** [primefaces/community#102] [PrimeVue] [answered]
I am using Tailwind in unstyled mode, using `usePassthrough` to extend the default Tailwind configuration. When defining a global passthrough, it's possible to set the `mergeProps` flag which will merge any new classes with the default classes. Is there a way of doing this on an individual basis, or does it have to be globally enabled?

Currently, if I define a passthrough directly on the component, it completely overwrites the existing styled instead of extending them, results can be seen below. In my test case, I would expect both to appear the same.

```
const CustomTailwind = usePassThrough(
    Tailwind,
    {
        panel: {
            title: {
                class: ['leading-none font-light text-2xl']
            }
        },
        button: ({props, context}: { props: ButtonProps, context: ButtonContext }) => ({
            root: {
                'class': [
                    {'bg-indigo-500 hover:bg-indigo-600 border-indigo-500': props.severity === 'info'}
                ]
            }
        })
    },
    {
        mergeSections: true,
        mergeProps: true
    }
);
```

```
<template>
  <Button label="Submit" icon="pi pi-check" severity="secondary"
          :pt="{
          root: {
            'class': [
              'bg-indigo-500 hover:bg-indigo-600 border-indigo-500'
           ]
          }
  }"
  />

  <Button label="Submit Default" icon="pi pi-check" severity="info"/>

</template>
```

![image](https://github.com/primefaces/community/assets/8271043/f37014be-e334-4c6e-9c9a-948cf9083aa7)

**Comment by tugcekucukoglu** [comment, marked as answer]
Same logic can be implemented with `ptOptions` property. For example

```
<Button
      label="Submit"
      :pt="{
        root: {
          class: ['bg-indigo-500 hover:bg-indigo-600 border-indigo-500'],
        },
      }"
      :ptOptions="{
        mergeSections: true,
        mergeProps: true,
      }"
    />
```

**Reply by willbdv** [reply]
Perfect, thank you.

**Comment by bweis** [comment]
This still won't work.  What is required is an external library such as tailwind-merge.  See my PR in the primereact repo for further info on this.  Right now class merging for tailwind is broken on all the primefaces repos from what I can tell.  https://github.com/primefaces/primereact/pull/5209

**Comment by tugcekucukoglu** [comment]
I want to share this Tailwind example https://stackblitz.com/edit/vitejs-vite-zz1pgb?file=src%2FApp.vue. 

Do you have a reproducer link so that we can see the broken part?

**Reply by bweis** [reply]
See example outlined here in primereact.

https://github.com/primefaces/primereact/pull/5215

Primevue should have the same limitations without an external merging library that accounts for tailwind's class "strength".
